sympy/solvers/ode.py

```diff
@@ -1335,77 +1335,59 @@ class in it.  Note that a hint may do this anyway if
                 matching_hints["Liouville_Integral"] = r
 
         # Homogeneous second order differential equation of the form
-        # a3*f(x).diff(x, 2) + b3*f(x).diff(x) + c3, where
-        # for simplicity, a3, b3 and c3 are assumed to be polynomials.
+        # a3*f(x).diff(x, 2) + b3*f(x).diff(x) + c3
         # It has a definite power series solution at point x0 if, b3/a3 and c3/a3
         # are analytic at x0.
         deq = a3*(f(x).diff(x, 2)) + b3*df + c3*f(x)
         r = collect(reduced_eq,
             [f(x).diff(x, 2), f(x).diff(x), f(x)]).match(deq)
         ordinary = False
```

Comment:
```
In [23]: eq = (x-2)**2*(f(x).diff(x, 2)) + (x-2)*(f(x).diff(x)) + ((x-2)**2 - 4)*f
    ...: (x)

In [24]: eq
Out[24]: 
           2                                               
       2  d                  d          ⎛       2    ⎞     
(x - 2) ⋅───(f(x)) + (x - 2)⋅──(f(x)) + ⎝(x - 2)  - 4⎠⋅f(x)
           2                 dx                            
         dx                                                

```

From here we need to match the equation and `match` is not able to match.
r is` None`.